### ISSUE TITLE
Website local development

### ISSUE THREAD

**Issue by AlexxNB** [inkstitch/inkstitch#1112]
Hello, I'm not familiar with Ruby ecosystem, but when I tried to start `bundle exec jekyll serve` (as mentioned on Jekyll's site) I got a couple of errors. I tried some other methods of Jekyll installations and now stop with error:
![изображение](https://user-images.githubusercontent.com/10389013/113386508-1400bb80-9393-11eb-9242-8c2aeae232bd.png)

I don't know the best way to avoid this. It will be cool if you will provide a Dockerfile with needed setup to easy run development environment.

I already done translation in Crowdin(please accept it!), now I would to translate Website, but have this issue.

**Comment by kaalleen** [comment]
First of all thank you for your work on the translations. Great job! I approved your translations and they will be added to Ink/Stitch soon (as we are close to the next release).

Now to your jekyll issue... Have you also updated your gem file to use the minimal mistakes theme?
https://mmistakes.github.io/minimal-mistakes/

**Comment by AlexxNB** [comment]
Great, thanks for approving - I'll fix translation when see how it will look in the extension(I suppose many strings won't fit their places). 

And thank you for your project - it s cool to have it, because I live in linux for a long time already and other alternatives are horrible to me.

As I wrote - Ruby ecosystem is far from me(I have no idea what is mean "gem file", for example). I need read  a lot of info that I don't need in my other projects. Dockerfile in the repo where is fully ready Jekill with all needed additions will be great for developers who don't want to know a lot about Jekill, but just fix text. 

Now I tried command from your link and another error:
![изображение](https://user-images.githubusercontent.com/10389013/113410230-956c4400-93bb-11eb-8bc3-0e2bb16db46c.png)

I'll try to use any docker image with Jakill to setup it its environment, maybe it will help me.

**Comment by AlexxNB** [comment]
Ok, I found that I need run `gem install "minimal-mistakes-jekyll"` instead `gem "minimal-mistakes-jekyll"`. But then I stuill have error from the first message about `Could not locate the included file 'video'...`

**Comment by AlexxNB** [comment]
Looks like it doesn't know what to do with `{% include video id="qXntE1X1RIw" provider="youtube" %}` in posts. Should I install something else to handle this?

**Comment by kaalleen** [comment]
The minimal mistakes theme should take care of it. So I assume it is still not fully running.
Sorry, I didn't mean to make it hard for you to get it running. I don't know how to setup a system to run it easily.

When I find the time I will make the website translatable ready for your language. There are some things need to be done first.

If you don't mind I can send you a request to join the Ink/Stitch repository. Then you can update the files directly onto the website.

**Comment by lexelby** [comment]
I like the idea of a Dockerfile.

**Comment by AlexxNB** [comment]
> If you don't mind I can send you a request to join the Ink/Stitch repository. Then you can update the files directly onto the website.

No, editing files without local copy of the site in the browser is bad idea =)

Anyway, I have success and write Dockerfile to easy start site without any troubles above:
```dockerfile
FROM jekyll/jekyll:4.2.0

RUN mkdir /gem && chmod 777 /gem && cd /gem \
 && echo 'source "https://rubygems.org"' >> Gemfile \
 && echo 'gem "github-pages", group: :jekyll_plugins' >> Gemfile \
 && echo 'gem "jekyll-include-cache", group: :jekyll_plugins' >> Gemfile \
 && bundle

WORKDIR /srv/jekyll

VOLUME /srv/jekyll

EXPOSE 4000

ENTRYPOINT cp /gem/Gemfile* /srv/jekyll/ && jekyll serve --livereload
```
It should be placed in site branch root and then run following commands:

```bash
# build image locally (do it once)
docker build -t inkstich-site .

# run site in development mode with livereload
docker run --rm --name Inkstitch --volume="$PWD:/srv/jekyll" -p 4000:4000 -it inkstitch-site
```
Then you will be able to open browser on https://localhost:4000 and see the local copy of the site. Change and save any source file to see changes (rebuilding take about 1 minute =( )

I could make PR to add Dockerfile to the repo with some description in Readme

**Comment by kaalleen** [comment]
That'd be great. Maybe then we will have more users translating the website.

I updated the website to be translatable ready for russian language. Internal links on the pages will still lead to the english version as I haven't updated them. Make sure you pull the newest version to your local build ;)

**Comment by AlexxNB** [comment]
Ok, great! Will try it now.
Any plans to migrate to other static site generator? Jekyll looks very unfriendly and waiting 60-100 seconds on every page save is PAIN =) i18n managing also looks very complicated.
I can lookup for better option and propose something.

**Comment by AlexxNB** [comment]
In newest version was changed this in _config.yml:
```yaml
theme                   : "minimal-mistakes-jekyll"
#remote_theme           : "mmistakes/minimal-mistakes"
```
In this case Dockerfile changes also:
```dockerfile
FROM jekyll/jekyll:4.2.0

RUN gem install "minimal-mistakes-jekyll"

WORKDIR /srv/jekyll

VOLUME /srv/jekyll

EXPOSE 4000

ENTRYPOINT jekyll serve --livereload
```
It works better - now rebuilding is about 10 seconds =)

**Comment by AlexxNB** [comment]
And from the start we have a problem.

![изображение](https://user-images.githubusercontent.com/10389013/113429237-de82bf00-93e0-11eb-9e8b-74d476ef427a.png)

As you can see font of russian text and english one is quite different. This happens because font Barlow doesn't have russian glyphs and browser use default font on the machine.  

Is it possible in Jekyll to use custom font for specified language? I can found similar font with cyrillic symbols support.

**Comment by kaalleen** [comment]
Sorry, the update of the theme was by accident and is causing the issue. I shouldn't do things too fast, sorry.

**Comment by kaalleen** [comment]
Ok, I updated the branch to return to the orignal theme setting. The other one I only use locally.

If you run `bundle exec jekyll serve --incremental` it will not take long to update the site.

**Comment by kaalleen** [comment]
Yes, I am sure we can do that. Tell me which font you want to use instead. When choosing a font, make sure it's license will fit our needs. We used Barlow here, because of our choice in the print pdf output.

**Comment by AlexxNB** [comment]
> Sorry, the update of the theme was by accident and is causing the issue. I shouldn't do things too fast, sorry.

Heh, it is sad, because approach with `theme` instead `remote theme` works much better and faster.

> If you run bundle exec jekyll serve --incremental it will not take long to update the site.

I already tried this before, it doesn't work. For example it does not rebuild when I change `navigation_xx.yml`.  So with `remote theme` I still need wait 60-100 seconds each time. 

> Yes, I am sure we can do that.

Looks like there are two fonts on the site now Barlow and Barlow Condensed. I suggest to replace they in russian version with [Fira Sans and Fira Sans Extra Condensed](https://fonts.google.com/?query=fira) accordingly.

They looks pretty similar except couple glyphs:

![изображение](https://user-images.githubusercontent.com/10389013/113446661-ec483c80-9400-11eb-923b-7898ad5f952d.png)

**Comment by AlexxNB** [comment]
I just tried site with Fira Sans fonts and it looks good.  Font size in content part a little bit bigger, but readability much better.
![изображение](https://user-images.githubusercontent.com/10389013/113473801-89908880-9474-11eb-97a5-bb18901a92f3.png)

I have no idea how to apply font only for russian part of the site, so in development I use font globally.

What is my workflow? 
* I should translate whole site and the PR changes then?
* I should do translation partially and PR each part?
* I should ask to be contributor in the repo and commit translated files directly into gh-pages branch?

---

And a question with app translation. What is the easiest way to bring translation from repo to  my installed Ink/Stitch?

**Comment by kaalleen** [comment]
> I have no idea how to apply font only for russian part of the site, so in development I use font globally.

My idea is to set the lang attribute in the html tag correctly, which will then make it easy to distinguish which fonts should be used. I'll take care of that.

Note when starting your translations, that we will have a new release soon. This we are very close I uploaded a few changes already, so you will not have to do the work twice. I did not update the installation process yet. So it you might want to wait with this one.

> I should translate whole site and then PR changes?
> I should do translation partially and PR each part?

You can decide that on your own. But since I uploaded everything and in english you can do it partially as well.

> I should ask to be contributor in the repo and commit translated files directly into gh-pages branch?

Yes, that makes sense. I'll send you an invitation.

**Comment by AlexxNB** [comment]
> My idea is to set the lang attribute in the html tag correctly, which will then make it easy to distinguish which fonts should be used. I'll take care of that.

Try this fonts in English site too. Maybe it makes sense to use it globally by default?
* It has not only Cyrillic part but glyphs for some other languages. 
* Export to PDF also should support Cyrillic an other languages.
* Content part looks really better with this font.
* It is pretty similar with Barlow font, so difference will be minimal

TIP: use `woff` font format instead `ttf`, it may help to reduce size of the font files.

**Comment by kaalleen** [comment]
I'll have to test it. The font decision was made long ago. Maybe there are some other people who want to comment on that.

I'll did send you an invitation. But I cannot work on the website today since I've got to go to work now. So I'll test it later.

For the theme issues... You can still run the local theme on your computer (I do too). It's just that the repository will need the remote theme to update the website correctly.

**Comment by AlexxNB** [comment]
> Note when starting your translations, that we will have a new release soon. This we are very close I uploaded a few changes already, so you will not have to do the work twice. I did not update the installation process yet. So it you might want to wait with this one.

Ok, I'll wait for new version release, it will be easier when navigation structure will not change rapidly. Please notify me when contribution will be safe.

> I'll did send you an invitation.

Thanks, accepted it. 

Is there any live discussion?  Discord or Telegram maybe?

**Comment by lexelby** [comment]
I don't think anyone has set up a discord or anything yet.  We do have GitHub discussions enabled.  I kind of like discussing things here so that we have a record, and other people can read what was said more easily.

**Comment by kaalleen** [comment]
> Ok, I'll wait for new version release, it will be easier when navigation structure will not change rapidly. Please notify me when contribution will be safe.

Ink/Stitch 2.0.0 is out. There still may be minor changes to make, but I'd say you are good to go!

**Comment by AlexxNB** [comment]
Oh nice. Great, congrats!
I'll start translating soon, right now I should do my work...
